Core/GDCore/Events/Expression.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+
+#include "GDCore/Events/Expression.h"
+
+#include "GDCore/Events/Parsers/ExpressionParser2.h"
+#include "GDCore/String.h"
+
+namespace gd {
+
+Expression::Expression() : node(nullptr){};
+
+Expression::Expression(gd::String plainString_)
+    : node(nullptr), plainString(plainString_){};
+
+Expression::Expression(const char* plainString_)
+    : node(nullptr), plainString(plainString_){};
+
+Expression::Expression(const Expression& copy)
+    : node(nullptr), plainString{copy.plainString} {};
+
+Expression& Expression::operator=(const Expression& expression) {
+  plainString = expression.plainString;
+  node = nullptr;
+  return *this;
+};
+
+Expression::~Expression(){};
+
+std::unique_ptr<ExpressionNode> Expression::GetRootNode(
+    const gd::String& type, gd::ExpressionParser2& parser) const {
```

Comment:
Is there a risk that the parser being passed changes compared to the "memoized" node? We should at least log this, to see when and why it happens, and what are the risks.

Reply:
(and we could if necessary memoize according to the parser address in memory or the content of the parser, like the objects container, but this can be risky too)